src/games/gameitem.py

```diff
@@ -218,99 +218,26 @@ def tooltip(self, data):
-        rows = [self._player_table_row(player, align) for player in team]
-        return team_table_start + "".join(rows) + team_table_end
-
-    def _player_table_row(self, player, align):
-        if isinstance(player, str):
-            country = "<td></td>"
-        else:
-            country = "<td>{country_icon}</td>"
-        pname = ("<td align='{alignment}' valign='middle' width='135'>"
-                 "{player}"
-                 "</td>")
-        order = [pname, country] if align == 'right' else [country, pname]
-        player_row = "<tr>{}{}</tr>".format(*order)
-
-        if isinstance(player, str):
-            return player_row.format(alignment=align, player=player)
-        else:
-            return player_row.format(
-                country_icon=self._country_icon_fmt(player),
-                alignment=align,
-                player=self._player_fmt(player))
-
-    def _country_icon_fmt(self, player):
-        icon_path_fmt = os.path.join("chat", "countries", "{}.png")
-        icon_path = icon_path_fmt.format(player.country.lower())
-        icon_abs_path = os.path.join(util.COMMON_DIR, icon_path)
-        return "<img src='{}'>".format(icon_abs_path)
-
-    def _player_fmt(self, player):
-        if player == self._me.player:
-            pformat = "<b><i>{}</b></i>"
-        else:
-            pformat = "{}"
-        player_string = pformat.format(player.login)
-        if player.rating_deviation < 200:   # FIXME: magic number
-            player_string += " ({})".format(player.rating_estimate())
-        return player_string
-
-    def _observers_tooltip(self, observers):
-        if not observers:
-            return ""
-
-        observer_fmt = "{country_icon} {observer}"
-
-        observer_strings = [observer_fmt.format(
-            country_icon=self._country_icon_fmt(observer),
-            observer=observer.login)
-            for observer in observers]
-        return "Observers: " + ", ".join(observer_strings)
-
-    def _mods_tooltip(self, mods):
-        if not mods:
-            return ""
-        return "<br/>With: " + "<br/>".join(mods.values())
 
     def format(self, teams, observers, mods):
-        teamtip = self._teams_tooltip(teams)
-        obstip = self._observers_tooltip(observers)
-        modtip = self._mods_tooltip(mods)
-
-        if modtip:
-            modtip = "<br/>" + modtip
-
-        return self.TIP_FORMAT.format(teams=teamtip,
-                                      observers=obstip,
-                                      mods=modtip)
+        def transform_playercountry_to_path(player):
+            if player is None or isinstance(player, str):
+                return
+            else:
+                icon_path_fmt = os.path.join("chat", "countries", "{}.png")
+                icon_path = icon_path_fmt.format(player.country.lower())
+                icon_abs_path = os.path.join(util.COMMON_DIR, icon_path)
+                player.countrypath = icon_abs_path
```

Comment:
Please don't set extra attributes on classes. Only classes' methods should do that, otherwise we get a real mess. Make the `country_to_path` method receive a country name and return a filepath, then make a dict comprehension, something like this:
```
flags = {p.country: self.country_to_filepath(p.country) for p in team for team in teams}
flags.update({p.country: self.country_to_filepath(p.country) for p in observers})
```
Or construct the dict via regular iteration. Then pass these flags as a separate argument to jinja.

Reply:
okay, didnt knew that, so you mean instead of putting it into the player class, i take sth like the login and the country, put it into a dict, pass the country path + login into the template and in the template get the path per-playerlogin? sounds doable for me

Reply:
You won't even need the login - just make a country -> file mapping and access it in jinja like you would in python - something like `dict[player.country]`.